### PR TITLE
[S3 Repository] Change default retry mechanism of s3 clients to Standard Mode

### LINKED ISSUE #15397: [Remote State] Add jitter to downloads from remote store
### Is your feature request related to a problem? Please describe

For remote cluster state publication, all the entity metadata in the cluster state are uploaded as separate files. If there are multiple entities in the cluster state that are changed, all these will be uploaded by the active cluster manager and downloaded at the same time by the follower nodes. This sudden burst in traffic to backing remote storage could cause throttling. 

### Describe the solution you'd like

Add jitter while downloading remote state files to mitigate the risk of throttling.

### Related component

Cluster Manager

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Arpit-Bandejiya**
### Description
This PR change the Default Backoff Strategy to `Standard` from `Legacy` . More info about it here: https://docs.aws.amazon.com/sdkref/latest/guide/feature-retry-behavior.html 

### Related Issues
Resolves #15397

### Check List
- [x] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/common/remote/AbstractRemoteWritableEntityManager.java [hunk 1]
```diff
@@ -24,6 +29,14 @@ public abstract class AbstractRemoteWritableEntityManager implements RemoteWrita
      */
     protected final Map<String, RemoteWritableEntityStore> remoteWritableEntityStores = new HashMap<>();
 
+    /**
+     * RemoteClusterStateSettings to add jitter in the download calls.
+     * Intentionally set to null to make sure the inheriting class configure it correctly.
+     */
+    protected RemoteClusterStateSettings remoteClusterStateSettings = null;
```
Comment (himshikha) [inline server/src/main/java/org/opensearch/common/remote/AbstractRemoteWritableEntityManager.java:36]: This doesn't ensure inheriting classes will override this value. Should we instead have an abstract method that inheriting classes will need to implement?

File: server/src/main/java/org/opensearch/common/remote/AbstractRemoteWritableEntityManager.java [hunk 2]
```diff
@@ -24,6 +29,14 @@ public abstract class AbstractRemoteWritableEntityManager implements RemoteWrita
...
+
+    protected final Random random = DefaultRandomObject.INSTANCE;
```
Comment (himshikha) [inline server/src/main/java/org/opensearch/common/remote/AbstractRemoteWritableEntityManager.java:38]: Is this being used?

File: server/src/main/java/org/opensearch/gateway/remote/DefaultRandomObject.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import org.opensearch.common.Randomness;
+
+import java.util.Random;
+
+/**
+ * Utility to provide a {@link Random} static instance
+ *
+ * @opensearch.internal
+ */
+public class DefaultRandomObject {
```
Comment (himshikha) [inline server/src/main/java/org/opensearch/gateway/remote/DefaultRandomObject.java:20]: 1. We can move this to commons package. 
2. Nit: Naming can be RandomLong if the intent is to have one instance of type long only here.

**Review by Bukhtawar — COMMENTED** [review]
Why can't we bake jitter in the remote store clients like S3/GCP utilising SDK client configurations?

File: server/src/main/java/org/opensearch/common/remote/RemoteWriteableEntityBlobStore.java [hunk 4]
```diff
@@ -89,6 +94,17 @@ public void readAsync(final U entity, final ActionListener<T> listener) {
...
+        Runnable runnable = () -> {
+            try {
+                listener.onResponse(read(entity));
+            } catch (Exception e) {
+                listener.onFailure(e);
+            }
+        };
+        threadPool.scheduleUnlessShuttingDown(TimeValue.timeValueMillis(delayMillis), executor, runnable);
+    }
+
```
Comment (Bukhtawar) [inline server/src/main/java/org/opensearch/common/remote/RemoteWriteableEntityBlobStore.java:107]: Shouldn't this jitter handling be done on the remote store side, by making it configurable at the repository level

Comment (Arpit-Bandejiya) [reply server/src/main/java/org/opensearch/common/remote/RemoteWriteableEntityBlobStore.java]: Removed this piece of code.

**Review by gbbafna — APPROVED** [review]

**Review by Bukhtawar — COMMENTED** [review]
What about sync clients? Do we need to change there as well?

**Review by Bukhtawar — APPROVED** [review]

**Review by Bukhtawar — APPROVED** [review]

**Review by gbbafna — APPROVED** [review]

**Pull request merged**